src/common/lpspi/bus/eh1_impl.rs

```diff
@@ -0,0 +1,81 @@
+use cassette::Cassette;
+
+use crate::lpspi::transfer_actions::{
+    create_actions_read_write, create_actions_read_write_in_place,
+};
+
+use super::{Lpspi, LpspiError};
+
+impl<const N: u8> eh1::spi::ErrorType for Lpspi<'_, N> {
+    type Error = LpspiError;
+}
+
+impl<const N: u8, T> eh1::spi::SpiBus<T> for Lpspi<'_, N>
+where
+    T: crate::lpspi::LpspiWord,
+{
```

Comment:
All of these implementations use `cassette` to spin on the future until it resolves. `cassette` doesn't care about wakers; it's just spinning, calling `poll()`, which calls a sub-future's `poll()`, which calls a sub-future's `poll()`, which calls... 

Those `polls()` conclude with a call to `StatusWatcherFuture`. This custom future takes a critical section every time we `poll()` it.

My concern is that by spinning on the future, we're frequently entering and exiting a critical section while we're driving the state machine. Could this happen?

Reply:
Yes, that does indeed happen. But that shouldn't be a problem because the critical sections are very short.

Reply:
Is that a problem?

Reply:
> I also don't see how taking these critical sections during blocking I/O adds value to the user's system

It doesn't. Yes, it would be better to for several reasons to provide a dedicated blocking API. Polling the async function was purely to reduce implementation complexity.

Reply:
Actually, similar to the waker problem, if we know we can only ever get called by a single context, we could get away with a `TryLock` and an `Atomic<Option<Waker>>`. I might give that a try.

Reply:
I kind of expected the whole critical section thing to compile to two asm instructions, one disable interrupts instruction and one enable interrupts instruction. Where does the complexity come from?

Reply:
Correction, I just realized that interrupts also operate in this. So it's not quite as straight forward.